### ISSUE TITLE
PR / Issue organization

### ISSUE THREAD

**Issue by zertosh** [substack/node-browserify#1232]
How about we borrow some of [these](https://github.com/facebook/react/labels) labels?

I think these are useful:
```
GH: Review-needed <-- Another collaborator needs to look at it (but one already did)
GH: Needs-revision  <-- Something wrong with a PR that the author should fix
GH: Accepted   <-- Ready to be merged for the next release

Duplicate
Invalid
Wontfix

Reconsider <-- Kinda like a hopeful "wontfix"
Bug (already exists) 
Feature Request (already exists)
Question (already exists as Support)
```

---

Also, the issue backlog is pretty long. I noticed both of you have been going through the old stuff. I think we should focus on the newer stuff since those ppl probably moved on or found a workaround. I propose that we declare "issue bankruptcy" on almost everything from before 2015.

cc: @terinjokes @jmm

**Comment by terinjokes** [comment]
With just three of us, I'm not sure we need JIRA-style labels. We can probably just get by with seeing the activity on the tracker and mentions, as we're doing right now :~).

>  I propose that we declare "issue bankruptcy" on almost everything from before 2015.

Most of the issues I've looked at so far are still relevant; the bug is still present, the feature request still useful, the API still acting in confusing ways

**Comment by zertosh** [comment]
> JIRA-style labels

omg. that's exactly what it is. what am i becoming!?!?!?!

> We can probably just get by with seeing the activity on the tracker and mentions, as we're doing right now

Oh no, the labels are mostly for users, not us.

> Most of the issues I've looked at so far are still relevant; the bug is still present,

I don't doubt this for a min. I just don't want to neglect users who have for sure have an immediate issue as opposed to those that may have moved on.

> Amusingly, opening an issue to talk about coordination was on my task list for today.

:smile:

**Comment by jmm** [comment]
FWIW, I don't have strong feelings either way about the additional labels.

As far as older issues, I don't know what the ideal solution is, but I think it's very beneficial to respond to them in a way that doesn't alienate the posters, especially the more "legit" (I'm not even going to attempt to define this, but I'm sure you get the gist) or high quality the reports are. It's too bad that there are some reports that have been there for quite some time without even being acknowledged.

Maybe as a first pass it would make sense to acknowledge and close older (6 months maybe?) issues that don't have sufficient information to reproduce, with encouragement to open a new issue with more information if it's still a problem.

FWIW I created a wiki to refer people to for guidance on posting higher quality reports:
[How to report issues with browserify](https://github.com/substack/node-browserify/wiki/How-to-report-issues-with-browserify). I was starting to repeat a lot of the same things, like format code, so I created this so I (or whoever) could link to the repetitive stuff and just refer to the relevant sections.

**Comment by terinjokes** [comment]
> I'm not even going to attempt to define this, but I'm sure you get the gist

I believe I've understood.

> Maybe as a first pass it would make sense to acknowledge and close older (6 months maybe?) issues that don't have sufficient information to reproduce

I've been traveling the last week or so, so I've started to do something similar when I find myself with 30-45 minutes of free time. I don't think most previous issues have magically gone away, and thus see it was a disservice to the community to not even acknowledge that they might exist.

> It's too bad that there are some reports that have been there for quite some time without even being acknowledged.

:cry: 

> FWIW I created a wiki to refer people to for guidance on posting higher quality reports: How to report issues with browserify.

GitHub will provide a link to CONTRIBUTING.md on the "New Issue" screen if it exists in the repo, so it might be worth it to add as a section, alongside how to develop for Browserify and it's dependencies. (I'm not sure how many would-be bug reporters click through to read it).

**Comment by zertosh** [comment]
> I'm not even going to attempt to define this, but I'm sure you get the gist
 
Gotcha too.

> Maybe as a first pass it would make sense to acknowledge and close older (6 months maybe?) issues that don't have sufficient information to reproduce, with encouragement to open a new issue with more information if it's still a problem.

That seems like a fair compromise.

> I created a wiki to refer people to for guidance on posting higher quality reports:
How to report issues with browserify.

I've seen you linking people there and it rocks! Awesome idea!

**Comment by jmm** [comment]
@terinjokes 
> GitHub will provide a link to CONTRIBUTING.md on the "New Issue" screen if it exists in the repo, so it might be worth it to add as a section, alongside how to develop for Browserify and it's dependencies. (I'm not sure how many would-be bug reporters click through to read it).

Thanks for the suggestion, that's a good idea. I've seen that before but didn't even think of it. I think I'll do that eventually. Like you said, who knows how many people click through though. That particular document will probably mostly be useful to link people to after they opened a ticket that has issues. I wish there were an easier way to link to wiki documents (some shorthand that would just link to it with the title).

@zertosh 
> That seems like a fair compromise.

I didn't elaborate, but I guess what I'm thinking is for issues newer than the cutoff (whatever it is) that lack sufficient information to reproduce and haven't been replied to, reply asking for more information and then wait a bit before closing and encouraging to open a new issue with more info if it's still a problem. Just thinking out loud.

**Comment by terinjokes** [comment]
@jmm If you tag an issue as "support" while asking for more information, I can run through and close out issues that haven't had a reply after a week.

**Comment by jmm** [comment]
@terinjokes Thanks, sounds good. Circling back to @zertosh's suggestion of creating labels, I was thinking that maybe a `needs-feedback` label would be helpful for those situations, to easily find those issues and expire them after some period of time so they're not open in an un-actionable state forever.

**Comment by terinjokes** [comment]
does "repro-please" sound good?

**Comment by zertosh** [comment]
:+1: for "repro-please"

**Comment by jmm** [comment]
Meta. Works for me. I'm even going to amuse myself by pronouncing it like a simular phrase that I won't repeat (but that can be heard in the song *Ether*).

**Comment by jmm** [comment]
Actually, now that I think about it, does `repro-please` encompass everything that `needs-feedback` would? What if, say, someone provides a test case that repros it for them, but it's platform specific and you need more information to track it down, or they don't say what version of browserify they were using, or whatever.

**Comment by terinjokes** [comment]
@jmm I think all the details needed to reproduce it falls under that label.

**Comment by jmm** [comment]
@terinjokes Ok.

**Comment by zertosh** [comment]
This is settled then :+1:

**Comment by jmm** [comment]
@zertosh @terinjokes This is just an FYI that I created the label `path-matching` for tracking certain  related issues that I would like to see squashed. And since GitHub unfortunately doesn't have a built-in mechanism for describing labels I created a [wiki](https://github.com/substack/node-browserify/wiki/Labels) for that purpose.

I'm not going to do a great job of explaining this right now, but there is a constellation of problems related to matching path and path-like values coming from various sources. See for example #1203, #1150. #1150 is a case where doing `browserify('./rel/path/file')` and `browserify('/abs/path/file`) both include `file` in the bundle, but the relative path version breaks the link to it as a dep of another file, producing a broken bundle.
